ext/date/php_date.c

```diff
@@ -311,16 +311,18 @@ static void date_throw_uninitialized_error(zend_class_entry *ce)
 {
 	if (ce->type == ZEND_INTERNAL_CLASS) {
 		zend_throw_error(date_ce_date_object_error, "Object of type %s has not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name));
-	} else {
-		zend_class_entry *ce_ptr = ce;
-		while (ce_ptr && ce_ptr->parent && ce_ptr->type == ZEND_USER_CLASS) {
-			ce_ptr = ce_ptr->parent;
-		}
-		if (ce_ptr->type != ZEND_INTERNAL_CLASS) {
-			zend_throw_error(date_ce_date_object_error, "Object of type %s not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name));
-		}
-		zend_throw_error(date_ce_date_object_error, "Object of type %s (inheriting %s) has not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name), ZSTR_VAL(ce_ptr->name));
+		return;
+	}
+
+	zend_class_entry *ce_ptr = ce;
+	while (ce_ptr && ce_ptr->parent && ce_ptr->type == ZEND_USER_CLASS) {
+		ce_ptr = ce_ptr->parent;
+	}
+	if (ce_ptr->type != ZEND_INTERNAL_CLASS) {
+		zend_throw_error(date_ce_date_object_error, "Object of type %s not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name));
+		return;
 	}
+	zend_throw_error(date_ce_date_object_error, "Object of type %s (inheriting %s) has not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name), ZSTR_VAL(ce_ptr->name));
```

Comment:
this also misses a `RETURN_THROWS()` (yes, I know it's the last line in the function, but still)

Reply:
I can add them for sure separately. I agree that the RETURN_THROWS() are useful